portfolio/src/main/webapp/script.js

```diff
@@ -135,23 +137,51 @@ async function loadMemes() {
     data.forEach(memeObject => {
         const memeDiv = document.createElement("div");
         const memeImg = document.createElement("img");
+        const icon = document.createElement("button");
         const memeDesc = document.createElement("p");
 
         memeImg.src = memeObject.url; 
         memeImg.classList.add("meme-image");
+        memeDesc.innerHTML = memeObject.desc
 
-        memeDiv.innerText = "[" + memeObject.timeStamp + "] " 
+        icon.innerHTML = '<i class="fa fa-trash-o"></i>';
+        icon.classList.add("trash");
+        icon.setAttribute( "onClick", 
+            'deleteThis("Meme", ' + memeObject.id + ')');
+        memeDiv.innerHTML = "[" + memeObject.timeStamp + "] " 
             + memeObject.author + ":";
-        memeDesc.innerHTML = memeObject.desc;
         memeDiv.appendChild(memeImg);
         memeDiv.appendChild(memeDesc);
+        memeDesc.appendChild(icon);
 
         memeDiv.classList.add("meme-div");
         memeContainer.appendChild(memeDiv);
 
     })
 }
 
+/** Delete this element. */
+async function deleteThis(elementType, elementId) {
+    let response = await fetch('/delete', {
+        method: 'POST',
+        headers: {
+            'Content-type': "application/json",
+        },
+        body: JSON.stringify({
+            type: elementType, 
+            id: elementId
+        }),
+    });
+}
+
+/** Delete all comments from the page and datastore. */
+async function deleteComments() {
+    let response = await fetch('/delete-data', {
+        method: 'POST',
+    });
+    response.then(getComments());
```

Comment:
why are you waiting for response to resolve before calling `getComments()`

i think you should be able to just call `getComments()` like before. What changed?

Reply:
Oh idk why this wasnt deleted, I meant to delete this, but anyways I was trying to make the html update instantly. I now realize that the code only relates to the backend and that I need to actually change the html.